### ISSUE TITLE
Replace "sleep 30" with terminus wait

### ISSUE THREAD

**Issue by hawkeyetwolf** [Lullabot/drainpipe#159]
The [Deploy to Pantheon](https://github.com/Lullabot/drainpipe/blob/main/scaffold/github/actions/pantheon/review/action.yml#L104) step calls `sleep 30` to make sure the code has finished deploying before running db updates, etc. Would it be worthwhile to try determining it programmatically? I have seen a few random failures during build, but not sure if it's related to this at all. 

The only thing I can think is to grep for `/Status +succeeded/` in the output below, which is not super clean, and not guaranteed to continue working as the terminus output could change. So maybe just close this issue as "won't fix" =D but figured I'd ask anyway.
```
❯ terminus workflow:info:status example-site.dev
 [notice] Showing latest workflow on example-site.
 -------------- -------------------------------------- 
  Workflow ID    f8d7261a-7508-11ed-bfd2-b2d290da5c19  
  Environment    dev                                   
  Workflow       Sync code on "dev"                    
  User           hawkeye@lullabot.com                  
  Status         running                               
  Started At     2022-12-06 01:54:51                   
  Finished At    1970-01-01 00:00:00                   
  Time Elapsed   11s                                   
 -------------- --------------------------------------
```
```
❯ terminus workflow:info:status example-site.dev
 [notice] Showing latest workflow on example-site.
 -------------- -------------------------------------- 
  Workflow ID    f8d7261a-7508-11ed-bfd2-b2d290da5c19  
  Environment    dev                                   
  Workflow       Sync code on "dev"                    
  User           hawkeye@lullabot.com                  
  Status         succeeded                             
  Started At     2022-12-06 01:54:51                   
  Finished At    2022-12-06 01:55:18                   
  Time Elapsed   26s                                   
 -------------- -------------------------------------- 
```

**Comment by hawkeyetwolf** [comment]
cc @deviantintegral in case you have other ideas!

**Comment by deviantintegral** [comment]
I think we can use one or more of these options to have reliable output formatting:

```
      --format=FORMAT   Format the result data. Available formats: csv,json,list,null,php,print-r,string,table,tsv,var_dump,var_export,xml,yaml [default: "table"]
      --fields=FIELDS   Available fields: Workflow ID (id), Environment (env), Workflow (workflow), User (user), Status (status), Started At (started_at), Finished At (finished_at), Time Elapsed (time) [default: ""]
      --field=FIELD     Select just one field, and force format to *string*. [default: ""]
```

**Comment by justafish** [comment]
@hawkeyetwolf good idea! :+1:

**Comment by deviantintegral** [comment]
I've actually done this on a client project and it works great. It also avoids surprising "Killed" output when drush is running and the container is restarted for the deployment if the time takes longer than 30 seconds (I was seeing 60). There's no need to capture the output; just run `ddev exec terminus build:workflow:wait ${{ inputs.site-name }}.${{ inputs.environment }}`.

And, the following in the Dockerfile in ddev to add it:

```
# Install build tools for features like waiting on workflows.
# Setting XDEBUG_MODE=off can be removed once the March 2023 release of ddev is
# out.
RUN sudo -u $username /bin/sh -c 'XDEBUG_MODE=off terminus self:plugin:install terminus-build-tools-plugin'
```

**Comment by deviantintegral** [comment]
I'm calling this a bug because as soon as it takes > 30 seconds to deploy everything breaks!

**Comment by baluv3** [comment]
We agreed on 5/3 to replace sleep 30 with terminus wait.

**Comment by justafish** [comment]
couple of other places this needs updating:
https://github.com/Lullabot/drainpipe/blob/main/scaffold/github/actions/pantheon/update/action.yml
https://github.com/Lullabot/drainpipe/blob/main/scaffold/gitlab/PantheonReviewApps.gitlab-ci.yml
